src/dpdk_statefulset_updater.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module used to update Kubernetes Statefulset to support DPDK."""
+
+import logging
+from typing import Iterable, Optional
+
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.models.core_v1 import Container
+from lightkube.resources.apps_v1 import StatefulSet
+
+logger = logging.getLogger(__name__)
+
+
+class DPDKStatefulSetUpdaterError(Exception):
+    """DPDKStatefulSetUpdaterError."""
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+
+class DPDKStatefulSetUpdater:
```

Comment:
```suggestion
class DPDKStatefulSet:
```

Reply:
How about something like `DPDKConfigurator`? It would look good with public methods called `is_configured` and `configure` you suggested below.

Reply:
Cool, I like that.
Done.